1.17.0: test suite is failing

Meson output:
<details>

```console
+ /usr/bin/meson setup --buildtype=plain --prefix=/usr --libdir=/usr/lib64 --libexecdir=/usr/libexec --bindir=/usr/bin --sbindir=/usr/sbin --includedir=/usr/include --datadir=/usr/share --mandir=/usr/share/man --infodir=/usr/share/info --localedir=/usr/share/locale --sysconfdir=/etc --localstatedir=/var --sharedstatedir=/var/lib --wrap-mode=nodownload --auto-features=enabled . x86_64-redhat-linux-gnu -D libportal=disabled -D docbook-docs=enabled
The Meson build system
Version: 1.2.0
Source dir: /home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0
Build dir: /home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu
Build type: native build
Project name: xdg-desktop-portal
Project version: 1.17.0
C compiler for the host machine: /usr/bin/gcc (gcc 13.2.1 "gcc (GCC) 13.2.1 20230728 (Red Hat 13.2.1-1)")
C linker for the host machine: /usr/bin/gcc ld.bfd 2.40-9
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (1.9.5)
Run-time dependency flatpak found: YES 1.12.2
Compiler for C supports arguments -Wno-unused-parameter: YES
Compiler for C supports arguments -Wno-sign-compare: YES
Compiler for C supports arguments -Wno-missing-field-initializers: YES
Checking for function "renameat2" : YES
Has header "sys/vfs.h" : YES
Has header "sys/mount.h" : YES
Has header "sys/statfs.h" : YES
Has header "sys/xattr.h" : YES
Has header "sys/extattr.h" : NO
Run-time dependency glib-2.0 found: YES 2.77.1
Run-time dependency gio-2.0 found: YES 2.77.1
Run-time dependency gio-unix-2.0 found: YES 2.77.1
Run-time dependency json-glib-1.0 found: YES 1.6.6
Run-time dependency fuse3 found: YES 3.15.1
Run-time dependency gdk-pixbuf-2.0 found: YES 2.42.10
Run-time dependency libgeoclue-2.0 found: YES 2.7.0
Dependency libportal skipped: feature libportal disabled
Run-time dependency libpipewire-0.3 found: YES 0.3.77
Run-time dependency libsystemd found: YES 251
Program bwrap found: YES (/usr/bin/bwrap)
Program xmlto found: YES (/usr/bin/xmlto)
Program gdbus-codegen found: YES (/usr/bin/gdbus-codegen)
Found pkg-config: /usr/bin/pkg-config (1.9.5)
Program gdbus-codegen found: YES (/usr/bin/gdbus-codegen)
Program gdbus-codegen found: YES (/usr/bin/gdbus-codegen)
Program gdbus-codegen found: YES (/usr/bin/gdbus-codegen)
Program glib-compile-resources found: YES (/usr/bin/glib-compile-resources)
Configuring xdg-desktop-portal.service using configuration
Configuring org.freedesktop.portal.Desktop.service using configuration
Configuring xdg-desktop-portal-rewrite-launchers.service using configuration
Program gdbus-codegen found: YES (/usr/bin/gdbus-codegen)
Configuring xdg-permission-store.service using configuration
Program gdbus-codegen found: YES (/usr/bin/gdbus-codegen)
Configuring xdg-document-portal.service using configuration
Configuring org.freedesktop.portal.Documents.service using configuration
Configuring org.freedesktop.impl.portal.PermissionStore.service using configuration
Configuring test.portal using configuration
Configuring limited.portal using configuration
Configuring org.freedesktop.portal.Documents.service using configuration
Configuring org.freedesktop.impl.portal.PermissionStore.service using configuration
Program pytest found: YES (/usr/bin/pytest)
Program python3 (dbus, dbusmock, gi) found: YES (/usr/bin/python3) modules: dbus, dbusmock, gi
Program python3 (xdist) found: YES (/usr/bin/python3) modules: xdist
tests/meson.build:308: WARNING: Project targets '>= 0.56.2' but uses feature introduced in '0.58.0': str.replace.
Program msgfmt found: YES (/usr/bin/msgfmt)
Program msginit found: YES (/usr/bin/msginit)
Program msgmerge found: YES (/usr/bin/msgmerge)
Program xgettext found: YES (/usr/bin/xgettext)
Configuring portal-docs.xml using configuration
Program rst2man found: YES (/usr/bin/rst2man)
Configuring config.h using configuration
Build targets in project: 65
WARNING: Project specifies a minimum meson_version '>= 0.56.2' but uses features which were added in newer versions:
 * 0.58.0: {'str.replace'}
NOTICE: Future-deprecated features used:
 * 0.62.0: {'pkgconfig.generate variable for builtin directories'}
 * 0.64.0: {'copy arg in configure_file'}

xdg-desktop-portal 1.17.0

  Directories
    DBus service dir            : /usr/share/dbus-1/services
    Flatpak interfaces dir      : /usr/share/dbus-1/interfaces/
    systemd user unit dir       : /usr/lib/systemd/user
    Installed tests dir         : /usr/libexec/installed-tests/xdg-desktop-portal

  Optional builds
    Enable docbook documentation: YES
    Enable libsystemd support   : YES
    Enable geoclue support      : YES
    Enable libportal support    : NO
    Enable installed tests:     : NO
    Enable python test suite    : YES

  User defined options
    auto_features               : enabled
    bindir                      : /usr/bin
    buildtype                   : plain
    datadir                     : /usr/share
    includedir                  : /usr/include
    infodir                     : /usr/share/info
    libdir                      : /usr/lib64
    libexecdir                  : /usr/libexec
    localedir                   : /usr/share/locale
    localstatedir               : /var
    mandir                      : /usr/share/man
    prefix                      : /usr
    sbindir                     : /usr/sbin
    sharedstatedir              : /var/lib
    sysconfdir                  : /etc
    wrap_mode                   : nodownload
    docbook-docs                : enabled
    libportal                   : disabled

Found ninja-1.11.1 at /usr/bin/ninja
```
</details>

And test suite is failing
<details>

```console
+ cd xdg-desktop-portal-1.17.0
+ /usr/bin/meson test -C x86_64-redhat-linux-gnu --num-processes 48 --print-errorlogs
ninja: Entering directory `/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu'
ninja: no work to do.
 1/28 xdg-desktop-portal / testdb                                     OK               0.01s   3 subtests passed
 2/28 xdg-desktop-portal / test-doc-portal                            SKIP             0.01s   0 subtests passed
 3/28 xdg-desktop-portal:portals / test-portals-account               OK               1.08s   1 subtests passed
 4/28 xdg-desktop-portal:portals / test-portals-background            OK               1.09s   1 subtests passed
 5/28 xdg-desktop-portal:portals / test-portals-camera                OK               1.08s   1 subtests passed
 6/28 xdg-desktop-portal:portals / test-portals-color                 SKIP             1.08s
 7/28 xdg-desktop-portal:portals / test-portals-email                 OK               1.09s   1 subtests passed
 8/28 xdg-desktop-portal:portals / test-portals-inhibit               OK               1.09s   1 subtests passed
 9/28 xdg-desktop-portal:portals / test-portals-location              OK               1.09s   1 subtests passed
10/28 xdg-desktop-portal:portals / test-portals-notification          OK               1.09s   1 subtests passed
11/28 xdg-desktop-portal:portals / test-portals-openfile              SKIP             1.10s
12/28 xdg-desktop-portal:portals / test-portals-openuri               OK               1.09s   1 subtests passed
13/28 xdg-desktop-portal:portals / test-portals-prepareprint          SKIP             1.09s
14/28 xdg-desktop-portal:portals / test-portals-print                 OK               1.10s   1 subtests passed
15/28 xdg-desktop-portal:portals / test-portals-savefile              SKIP             1.10s
16/28 xdg-desktop-portal:portals / test-portals-screenshot            OK               1.10s   1 subtests passed
17/28 xdg-desktop-portal:portals / test-portals-trash                 OK               1.10s   1 subtests passed
18/28 xdg-desktop-portal:portals / test-portals-wallpaper             OK               1.09s   1 subtests passed
19/28 xdg-desktop-portal:portals / limited-portals-openfile           SKIP             1.07s
20/28 xdg-desktop-portal:portals / limited-portals-savefile           SKIP             1.07s
▶ 21/28 /permissions/delete5 - xdg-desktop-portal:ERROR:../tests/test-permission-store.c:410:delete_permission_cb: assertion failed (error == NULL): GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying (g-dbus-error-quark, 4) FAIL
▶ 21/28                                                               ERROR
21/28 xdg-desktop-portal / test-permission-store                      ERROR            0.45s   killed by signal 6 SIGABRT
>>> XDP_UNINSTALLED=1 G_TEST_BUILDDIR=/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests XDG_DATA_DIRS=/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests/share MALLOC_PERTURB_=152 G_TEST_SRCDIR=/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/tests G_DEBUG=gc-friendly /home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests/test-permission-store
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
dbus-daemon[2300246]: [session uid=1000 pid=2300246] Activating service name='org.freedesktop.impl.portal.PermissionStore' requested by ':1.0' (uid=1000 pid=2300244 comm="/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1" label="kernel")
dbus-daemon[2300246]: [session uid=1000 pid=2300246] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
**
xdg-desktop-portal:ERROR:../tests/test-permission-store.c:410:delete_permission_cb: assertion failed (error == NULL): GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying (g-dbus-error-quark, 4)
cleaning up pid 2300246

(test program exited with status code -6)
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

22/28 xdg-desktop-portal / test-xdp-utils                             OK               0.01s   6 subtests passed
23/28 xdg-desktop-portal:pytest / pytest test_email                   FAIL             0.39s   exit status 4
>>> MALLOC_PERTURB_=215 /usr/bin/pytest --verbose --log-level=DEBUG -n auto -k test_email
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
ImportError while loading conftest '/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests/conftest.py'.
tests/conftest.py:57: in <module>
    def params() -> dict[str, Any]:
E   TypeError: 'type' object is not subscriptable
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

24/28 xdg-desktop-portal:pytest / pytest test_globalshortcuts         FAIL             0.39s   exit status 4
>>> MALLOC_PERTURB_=37 /usr/bin/pytest --verbose --log-level=DEBUG -n auto -k test_globalshortcuts
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
ImportError while loading conftest '/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests/conftest.py'.
tests/conftest.py:57: in <module>
    def params() -> dict[str, Any]:
E   TypeError: 'type' object is not subscriptable
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

25/28 xdg-desktop-portal:pytest / pytest test_inputcapture            FAIL             0.40s   exit status 4
>>> MALLOC_PERTURB_=82 /usr/bin/pytest --verbose --log-level=DEBUG -n auto -k test_inputcapture
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
ImportError while loading conftest '/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests/conftest.py'.
tests/conftest.py:57: in <module>
    def params() -> dict[str, Any]:
E   TypeError: 'type' object is not subscriptable
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

26/28 xdg-desktop-portal:pytest / pytest test_remotedesktop           FAIL             0.40s   exit status 4
>>> MALLOC_PERTURB_=26 /usr/bin/pytest --verbose --log-level=DEBUG -n auto -k test_remotedesktop
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
ImportError while loading conftest '/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests/conftest.py'.
tests/conftest.py:57: in <module>
    def params() -> dict[str, Any]:
E   TypeError: 'type' object is not subscriptable
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

27/28 xdg-desktop-portal:pytest / pytest test_clipboard               FAIL             0.44s   exit status 4
>>> MALLOC_PERTURB_=34 /usr/bin/pytest --verbose --log-level=DEBUG -n auto -k test_clipboard
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
ImportError while loading conftest '/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests/conftest.py'.
tests/conftest.py:57: in <module>
    def params() -> dict[str, Any]:
E   TypeError: 'type' object is not subscriptable
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

28/28 xdg-desktop-portal:pytest / pytest test_trash                   FAIL             0.41s   exit status 4
>>> MALLOC_PERTURB_=140 /usr/bin/pytest --verbose --log-level=DEBUG -n auto -k test_trash
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
stderr:
ImportError while loading conftest '/home/tkloczko/rpmbuild/BUILD/xdg-desktop-portal-1.17.0/x86_64-redhat-linux-gnu/tests/conftest.py'.
tests/conftest.py:57: in <module>
    def params() -> dict[str, Any]:
E   TypeError: 'type' object is not subscriptable
―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――


Summary of Failures:

21/28 xdg-desktop-portal / test-permission-store              ERROR            0.45s   killed by signal 6 SIGABRT
23/28 xdg-desktop-portal:pytest / pytest test_email           FAIL             0.39s   exit status 4
24/28 xdg-desktop-portal:pytest / pytest test_globalshortcuts FAIL             0.39s   exit status 4
25/28 xdg-desktop-portal:pytest / pytest test_inputcapture    FAIL             0.40s   exit status 4
26/28 xdg-desktop-portal:pytest / pytest test_remotedesktop   FAIL             0.40s   exit status 4
27/28 xdg-desktop-portal:pytest / pytest test_clipboard       FAIL             0.44s   exit status 4
28/28 xdg-desktop-portal:pytest / pytest test_trash           FAIL             0.41s   exit status 4

Ok:                 14
Expected Fail:      0
Fail:               7
Unexpected Pass:    0
Skipped:            7
Timeout:            0
```
</details>

```
    def params() -> dict[str, Any]:
E   TypeError: 'type' object is not subscriptable
```

These are because you're using Python 3.8 or older. The backwards-compatible syntax for older Pythons is `Dict[str, Any]`.

> `xdg-desktop-portal:ERROR:../tests/test-permission-store.c:410:delete_permission_cb: assertion failed (error == NULL): GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying (g-dbus-error-quark, 4)`

This one is (at least potentially) different: the permission store is crashing.

--- a/tests/meson.build
+++ b/tests/meson.build
@@ -273,7 +273,7 @@ python = pymod.find_installation(
   required: get_option('pytest'),
 )
 
-enable_pytest = pytest.found() and python.found()
+enable_pytest = pytest.found() and python.found() and python.language_version().version_compare('>=3.9')
 
 if enable_pytest
   subdir('templates')

